cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryConfig.java

```diff
@@ -65,21 +65,22 @@ public void validate() {
     Collection<RepositoryValidationFailure> failures = new ArrayList<>();
 
     if (provider == null) {
-      failures.add(new RepositoryValidationFailure("'provider' field cannot be null."));
+      failures.add(new RepositoryValidationFailure("'provider' field cannot be null or empty."));
     }
 
-    if (link == null) {
-      failures.add(new RepositoryValidationFailure("'link' field cannot be null."));
+    if (link == null || link.equals("")) {
+      failures.add(new RepositoryValidationFailure("'link' field cannot be null or empty."));
     }
 
-    if (defaultBranch == null) {
-      failures.add(new RepositoryValidationFailure("'defaultBranch' field cannot be null."));
+    if (defaultBranch == null || defaultBranch.equals("")) {
```

Comment:
saw some UI mocks where the branch was optional. Should we remove this check?

Reply:
bring @sumengwang with visibility. So in MVP we do everything on the default branch. If we don't specify it, it's gonna be a branch determined by git: https://stackoverflow.com/a/32503667

I think we should not make this field optional, at least for MVP.

Reply:
Offline discussed, the default branch could be the one set by Github if not specified otherwise